lib/bytes_by_prefix.c

```diff
@@ -62,12 +62,16 @@
 #define B_MULT	BSIZE		/* block size */
 #elif sgi
 #define B_MULT	BBSIZE		/* block size */
-#elif defined(__linux__) || defined(__sun) || defined(__hpux)
+#elif (defined(__linux__) || defined(__sun) || defined(__hpux)) && defined(DEV_BSIZE)
 #define B_MULT	DEV_BSIZE	/* block size */
```

Comment:
I'm unfamiliar with the other systems and don't know the right way to get rid of their ifdefs. Is this how you'd like it?

```c
#ifdef BSIZE
#define B_MULT BSIZE
#elif defined(BBSIZE)
#define B_MULT BBSIZE
#elif defined(DEV_BSIZE)
#define B_MULT DEV_BSIZE
#elif defined(UBSIZE)
#define B_MULT UBSIZE
#else
#warning Unable to find a good value for B_MULT, defaulting to 512
#define B_MULT 512
#endif
```